### PR TITLE
Pgdb dsl2

### PR DESCRIPTION
**Pull request by ypriverol**
## PR checklist

- [ ] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
 - [ ] If you've added a new tool - add to the software_versions process and a regex to `scrape_software_versions.py`
 - [ ] If you've added a new tool - have you followed the pipeline conventions in the [contribution docs](https://github.com/nf-core/pgdb/tree/master/.github/CONTRIBUTING.md)
 - [ ] If necessary, also make a PR on the nf-core/pgdb _branch_ on the [nf-core/test-datasets](https://github.com/nf-core/test-datasets) repository.
- [ ] Make sure your code lints (`nf-core lint .`).
- [ ] Ensure the test suite passes (`nextflow run . -profile test,docker`).
- [ ] Usage Documentation in `docs/usage.md` is updated.
- [ ] Output Documentation in `docs/output.md` is updated.
- [ ] `CHANGELOG.md` is updated.
- [ ] `README.md` is updated (including new tool citations and authors/contributors).

### CODE REVIEW

**Comment by daichengxin** [comment]
https://github.com/bigbio/pgdb/blob/6011982d86c85151158c75f5bd93924db0867b5c/workflows/pgdb.nf#L103
fix: `ensembl_downloader_config` -> `params.ensembl_downloader_config`

File: modules/local/cbioportalmutations/cbioportal_proteindb.nf [hunk 1]
```diff
@@ -0,0 +1,31 @@
+/**
+ * Generate proteinDB from cBioPortal mutations
+ */
+process CBIOPORTAL_PROTEINDB {
+
+    container "nfcore/pgdb:1.0.0"
```
Comment (ypriverol) [inline modules/local/cbioportalmutations/cbioportal_proteindb.nf]: @DongdongdongW Every step should using only the tools needs by the module, not the general (global container). In addition, we are only using mostly in all modules Docker, you need to add the Singularity and Conda containers.

File: workflows/pgdb.nf [hunk 2]
```diff
@@ -0,0 +1,232 @@
...
+    // Download data from ensembl for the particular species
+    ENSEMBL_FASTA_DOWNLOAD(params.ensembl_downloader_config)
+
+    ADD_REFERENCE_PROTEOME(ENSEMBL_FASTA_DOWNLOAD.out.ensembl_protein_database_sub)
+
+    //Concatenate cDNA and ncRNA databases
+    MERGE_CDNAS(ENSEMBL_FASTA_DOWNLOAD.out.ensembl_cdna_database_sub.collect(),ENSEMBL_FASTA_DOWNLOAD.out.ensembl_ncrna_database_sub.collect())
+
+    //Creates the ncRNA protein database
+    ADD_NCRNA(MERGE_CDNAS.out.total_cdnas,params.ensembl_config)
```
Comment (ypriverol) [inline workflows/pgdb.nf]: @DongdongdongW I have fixed most of the errors of the parameters but we must standarize across all modules the variables. 

Currently we are passing params to most of the modules, however in the original implementation you didn't add the params prefix. In some other cases we don't pass the params but we call them inside the module with the param prefix. What we need to figure it out is are we passing params of the workflow to the modules or we remove the parameter from the module and just called the param inside.

**Comment by ypriverol** [comment]
@DongdongdongW when the pipeline starts, you get the following warning: 
```

WARN: Found unexpected parameters:
* --hostnames: null
* --gnomad_file_url: gs://gnomad-public/release/2.1.1/vcf/exomes/gnomad.exomes.r2.1.1.sites.vcf.bgz
* --gencode_url: ftp://ftp.ebi.ac.uk/pub/databases/gencode/Gencode_human/release_19
* --decoy_enzyme: Trypsin
* --decoy_method: decoypyrat
* --decoy: true
* --decoy_prefix: Decoy_
* --final_database_protein: final_proteinDB.fa
* --af_field: 
* --cbioportal_filter_column: CANCER_TYPE
* --cbioportal_accepted_values: all
* --cbioportal_study_id: all
* --cosmic_cellline_name: all
* --cosmic_cancer_type: all
* --vcf_file: null
* --ensembl_name: meleagris_gallopavo
* --protein_decoy_config: /Users/yperez/IdeaProjects/github-repo/BDP/pgdb/conf/protein_decoy.yaml
* --cbioportal_config: /Users/yperez/IdeaProjects/github-repo/BDP/pgdb/conf/cbioportal_config.yaml
* --cosmic_config: /Users/yperez/IdeaProjects/github-repo/BDP/pgdb/conf/cosmic_config.yaml
* --ensembl_config: /Users/yperez/IdeaProjects/github-repo/BDP/pgdb/conf/ensembl_config.yaml
* --ensembl_downloader_config: /Users/yperez/IdeaProjects/github-repo/BDP/pgdb/conf/ensembl_downloader_config.yaml
* --cosmic_password: null
* --cosmic_user_name: null
* --add_stop_codons: true
* --minimum_aa: 6
* --clean_database: true
* --cosmiccelllines_mutations: null
* --cosmiccelllines_genes: null
* --cosmicmutations: null
* --cosmicgenes: null
* --add_reference: true
* --gnomad: false
* --ensembl: false
* --cosmic_celllines: false
* --cosmic: false
* --cbioportal: false
* --vcf: false
* --altorfs: false
* --pseudogenes: false
* --ncrna: false
- Ignore this warning: params.schema_ignore_params = "hostnames,gnomad_file_url,gencode_url,decoy_enzyme,decoy_method,decoy,decoy_prefix,final_database_protein,af_field,cbioportal_filter_column,cbioportal_accepted_values,cbioportal_study_id,cosmic_cellline_name,cosmic_cancer_type,vcf_file,ensembl_name,protein_decoy_config,cbioportal_config,cosmic_config,ensembl_config,ensembl_downloader_config,cosmic_password,cosmic_user_name,add_stop_codons,minimum_aa,clean_database,cosmiccelllines_mutations,cosmiccelllines_genes,cosmicmutations,cosmicgenes,add_reference,gnomad,ensembl,cosmic_celllines,cosmic,cbioportal,vcf,altorfs,pseudogenes,ncrna" 

```

I think we have a major problem in the way the DSL2 is getting the parameters because non of them are recognized. Try to tackle that.

**Pull request merged**